src/main/java/com/aws/greengrass/clientdevices/auth/configuration/CAConfiguration.java

```diff
@@ -0,0 +1,44 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package com.aws.greengrass.clientdevices.auth.configuration;
+
+import com.aws.greengrass.config.Topics;
+import com.aws.greengrass.util.Coerce;
+import lombok.Getter;
+
+import java.util.List;
+
+import static com.aws.greengrass.clientdevices.auth.ClientDevicesAuthService.CA_PASSPHRASE;
+import static com.aws.greengrass.componentmanager.KernelConfigResolver.CONFIGURATION_CONFIG_KEY;
+import static com.aws.greengrass.lifecyclemanager.GreengrassService.RUNTIME_STORE_NAMESPACE_TOPIC;
+
+@Getter
+public class CAConfiguration {
+    public static final String CERTIFICATE_AUTHORITY_TOPIC = "certificateAuthority";
+    public static final String CA_CERTIFICATE_URI = "certificateUri";
+    public static final String CA_PRIVATE_KEY_URI = "privateKeyUri";
+    public static final String CA_TYPE_TOPIC = "ca_type";
+    private String caPrivateKeyUri;
+    private String caCertificateUri;
+    private List<String> caTypeList;
+    private String caPassphrase;
+
+    /**
+     * Updates the CA configuration with the latest CDA config.
+     *
+     * @param cdaConfigTopics CDA service configuration topics
+     */
+    public synchronized void setCAConfiguration(Topics cdaConfigTopics) {
```

Comment:
Actually, this class still isn't immutable. Think about the cert rotation workflow - it's still possible for that to trigger during a config update which could result in us getting into a non-deterministic state.

I actually want you to re-instantiate this config class whenever there is a config change. Everything should be final